src/open_inwoner/openklant/services.py

```diff
@@ -79,3 +84,303 @@ def update_user_from_klant(klant: Klant, user: User):
+                                }
+                            )
+                        except Exception:
+                            logger.exception(
+                                "Unable to register identificatoren for partij"
+                            )
+
+            partij = organisatie
+
+        if not partij:
+            logger.error("Unable to create OpenKlant2 partij for user")
+            return
+
+        msg = (
+            f"{'created' if created else 'retrieved'} partij {partij['uuid']} for user"
+        )
+        system_action(msg, content_object=user)
+
+        return partij
+
+    def retrieve_digitale_addressen_for_partij(
+        self, partij: Partij
+    ) -> list[DigitaalAdres]:
+        if expand := partij.get("_expand"):
+            if digitale_adressen := expand.get("digitaleAdressen"):
+                return digitale_adressen
+
+        expand_partij = self.client.partij.retrieve(
+            partij["uuid"], params={"expand": ["digitaleAdressen"]}
+        )
+
+        if expand := expand_partij.get("_expand"):
+            if digitale_adressen := expand.get("digitaleAdressen"):
+                return digitale_adressen
+
+        # TODO: A missing _expand can mean there are no addresses.
+        # See: https://github.com/maykinmedia/open-klant/issues/243
+        return []
+
+    def filter_digitale_addressen_for_partij(
+        self,
+        partij: Partij,
+        *,
+        soortDigitaalAdres: str,
+        adressen: Iterable[DigitaalAdres] | None = None,
+    ) -> list[DigitaalAdres]:
+        if adressen is None:
+            adressen = self.retrieve_digitale_addressen_for_partij(partij)
+
+        return [
+            digitaal_adres
+            for digitaal_adres in adressen
+            if digitaal_adres["soortDigitaalAdres"] == soortDigitaalAdres
+        ]
+
+    def get_or_create_digitaal_adres(
+        self,
+        partij: Partij,
+        soortAdres: Literal["email", "telefoon"],
+        adres: str,
+    ) -> tuple[DigitaalAdres, bool]:
+        digitale_adressen = self.filter_digitale_addressen_for_partij(
+            partij, soortDigitaalAdres=soortAdres
+        )
+        for digitaal_adres in digitale_adressen:
+            if digitaal_adres["adres"] == adres:
+                return digitaal_adres, False
+
+        return (
+            self.client.digitaal_adres.create(
+                data={
+                    "adres": adres,
+                    "soortDigitaalAdres": soortAdres,
+                    "verstrektDoorPartij": {
+                        "uuid": partij["uuid"],
+                    },
+                    "verstrektDoorBetrokkene": None,
+                    "omschrijving": "OIP profiel",
+                }
+            ),
+            True,
+        )
+
+    def update_user_from_partij(self, partij: Partij, user: User):
+        update_data = {}
+
+        adressen = self.retrieve_digitale_addressen_for_partij(partij)
+
+        if len(
```

Comment:
What's the reason for checking `len` instead of just truthiness of `email_adressen` (and `phone_adressen` below)?